### PR TITLE
Migration of sale_cancel_reason to 14.0

### PR DESCRIPTION
**Pull request by psugne**
Migrating `sale_cancel_reason` [13.0](https://github.com/OCA/sale-workflow/tree/13.0/sale_cancel_reason) -> 14.0

Also made a [PR](https://github.com/OCA/sale-workflow/pull/1627) on OCA

Please check _1c71182_ commit

### CODE REVIEW

**Review by paulius-sladkevicius — APPROVED** [review]
:+1:

File: sale_cancel_reason/model/sale.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2013 Guewen Baconnier, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    cancel_reason_id = fields.Many2one(
+        "sale.order.cancel.reason",
```
Comment (DarkoNikolovski) [inline sale_cancel_reason/model/sale.py]: Shouldn't be used attribute names for all field's attributes? If yes there are few other places with the same issue.

File: sale_cancel_reason/wizard/cancel_reason.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    _name = "sale.order.cancel"
+    _description = __doc__
+
+    reason_id = fields.Many2one("sale.order.cancel.reason", required=True)
+
+    def confirm_cancel(self):
+        self.ensure_one()
+        act_close = {"type": "ir.actions.act_window_close"}
+        sale_ids = self._context.get("active_ids")
```
Comment (DarkoNikolovski) [inline sale_cancel_reason/wizard/cancel_reason.py]: Any reason why private context is used?

**Review by DarkoNikolovski — APPROVED** [review]

File: sale_cancel_reason/wizard/cancel_reason.py (hunk 2 above)
Comment (psugne) [reply sale_cancel_reason/wizard/cancel_reason.py]: @DarkoNikolovski seems I miss this part. Thanks for pointing.

**Pull request merged**